testsuite/classlibrary/TestSynthDesc.sc

```diff
@@ -0,0 +1,88 @@
+TestSynthDesc : UnitTest {
+	test_SynthDesc_read_restores_metadata_in_desc_and_def {
+		var package = this.setupMetadataTests;
+		var synthdesc;
+
+		if(package.error.notNil) {
+			this.failed(thisMethod, "Error writing SynthDef: '%'".format(package.error.errorString));
+			^this.cleanupMetadataTests(package);
+		};
+
+		synthdesc = SynthDesc.read(package.path, keepDefs: true)[\test_md];
+		if(synthdesc.isNil) {
+			this.failed(thisMethod, "SynthDesc was not read from disk");
+			^this.cleanupMetadataTests(package);
+		};
+
+		this.checkMetadata(synthdesc, package.metadata, SynthDesc);
+
+		this.cleanupMetadataTests(package);
+	}
+
+	test_SynthDescLib_read_restores_metadata_in_desc_and_def {
+		var package = this.setupMetadataTests;
+		var synthdesc;
+
+		if(package.error.notNil) {
+			this.failed(thisMethod, "Error writing SynthDef: '%'".format(package.error.errorString));
+			^this.cleanupMetadataTests(package);
+		};
+
+		synthdesc = SynthDescLib.read(package.dir +/+ "*.scsyndef").global;
+		synthdesc = synthdesc[\test_md];
```

Comment:
you should make `\test_md` a constant somewhere

Reply:
Done.